### PR TITLE
CI updates

### PR DESCRIPTION
**Pull request by TriplEight**
- `Swatinem/rust-cache` handles Rust cache better

### CODE REVIEW

**Comment by kpp** [comment]
There is no need to fix CI, it works: https://github.com/libp2p/rust-libp2p/pull/2227/checks

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,32 +55,21 @@ jobs:
 
     - name: Install Rust wasm32-wasi
       uses: actions-rs/toolchain@v1.0.7
+
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:58]: Seems like an unnecessary diff.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,32 +55,21 @@ jobs:
...
-      with:
-        path: ~/.cargo
-        key: cargo-home-${{ hashFiles('Cargo.toml') }}
-
-    - name: Cache cargo build
-      uses: actions/cache@v2.1.6
-      with:
-        path: target
-        key: wasm-cargo-build-target-${{ hashFiles('Cargo.toml') }}
+    - uses: Swatinem/rust-cache@v1
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:72]: Same as with the `checkout` action, specifying the full version will avoid a direct dependabot PR after this is merged.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
A few nits but otherwise a welcoming change from my end. Personally I've had a good experience with `rust-cache` so far!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -65,32 +55,21 @@ jobs:
...
+
       with:
         toolchain: stable
         target: wasm32-wasi
         override: true
 
     - name: Install a recent version of clang
       run: |
-        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-        echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main" >> /etc/apt/sources.list
```
[the remark this answers is not in the document]
Comment (TriplEight) [reply .github/workflows/ci.yml:76]: yeah, it's not needed anymore.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -54,7 +44,7 @@ jobs:
       with:
         access_token: ${{ github.token }}
 
-    - uses: actions/checkout@v2.3.4
+    - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (TriplEight) [reply .github/workflows/ci.yml]: you're right

**Review by thomaseizinger — APPROVED** [review]
Thanks, this looks good!

We will have to wait for @mxinden to merge this. He is currently on vacation though so that might take a little longer :)

**Review by mxinden — APPROVED** [review]
Thank you @TriplEight. Very much appreciate your help here! Good to have a GitHub CI expert around.

**Pull request merged**